diff --git a/Makefile b/Makefile
index cd08b35..adbd98f 100644
--- a/Makefile
+++ b/Makefile
@@ -19,3 +19,26 @@ $(objpfx)tst-timezone: src/tst-timezone.c
 clean:
 	rm -rf obj src/yearistype
 	for i in tzdata*/*; do rm -f src/`basename $$i`; done
+
+extras:
+	g++ scripts/signature.c++ -Wall -Werror -xc++ -o src/signature
+	scripts/iso8601.perl > src/iso8601
+	scripts/zone-list.sh > zone.list
+	scripts/zone-generate.sh
+	perl -w -s scripts/find-aliases.perl \
+		-zones=zone.list -signatures=signatures -md5sum=md5sum \
+		-links=zone.link -zonetab=src/zone.tab > zone.alias
+	cat zone.alias | scripts/create-test-case.perl > timezones.c++
+	for name in `cat zone.alias` ; do echo ?? +0+0 $name ; done > zone-and-aliases.tab
+	scripts/prepare-timed-data.perl \
+		--zonetab=src/zone.tab \
+		--signatures=signatures \
+		--mcc-main=data/MCC \
+		--distinct=data/distinct.tab \
+		--single=data/single.tab \
+		data/mcc-wikipedia-*.html \
+		--single-output=timed/single.data \
+		--distinct-output=timed/distinct.data \
+		--full-output=timed/olson.data \
+		--country-by-mcc-output=timed/country-by-mcc.data \
+		--zones-by-country-output=timed/zones-by-country.data
diff --git a/scripts/prepare-timed-data.perl b/scripts/prepare-timed-data.perl
index 6362fc8..f615db9 100755
--- a/scripts/prepare-timed-data.perl
+++ b/scripts/prepare-timed-data.perl
@@ -80,7 +80,7 @@ sub read_mcc_list_by_parameter
 # 1
 sub read_wiki
 {
-  my $re = qr/^mcc-wikipedia-\d{4}-\d{2}-\d{2}(-*)?\.html$/ ;
+  my $re = qr/^[\w\-\/]*mcc-wikipedia-\d{4}-\d{2}-\d{2}(-*)?\.html$/ ;
   my @wiki_list = sort grep { /$re/ } @ARGV ;
   die "not a single wikipage given on command line" unless @wiki_list ;
   my $wiki_file = $wiki_list[-1] ;
diff --git a/scripts/zone-generate.sh b/scripts/zone-generate.sh
index a3fe426..6ad76e3 100755
--- a/scripts/zone-generate.sh
+++ b/scripts/zone-generate.sh
@@ -4,10 +4,10 @@
 
 # This script has to be executed in the build directory
 
-zic=bin/zic
-yearistype=bin/yearistype
-signature=bin/signature
-leapseconds=leapseconds
+zic=src/zic
+yearistype=src/yearistype
+signature=src/signature
+leapseconds=src/leapseconds
 
 output=zones
 signatures=signatures
@@ -20,10 +20,10 @@ test -x $signature
 test -r $leapseconds
 rm -rf $output $signatures $md5sum && mkdir -p $output
 
-input="africa antarctica asia australasia europe northamerica southamerica"
-input="$input iso8601"
-input="$input etcetera factory systemv backward"
-input="$input solar87 solar88 solar89"
+input="src/africa src/antarctica src/asia src/australasia src/europe src/northamerica src/southamerica"
+input="$input src/iso8601"
+input="$input src/etcetera src/factory src/systemv src/backward"
+input="$input src/solar87 src/solar88 src/solar89"
 
 for i in $input ; do
   echo "Processing '$i'"
diff --git a/scripts/zone-list.sh b/scripts/zone-list.sh
index 5295af9..fc71c20 100755
--- a/scripts/zone-list.sh
+++ b/scripts/zone-list.sh
@@ -4,8 +4,8 @@
 
 # This script has to be executed in the build directory
 
-zic=bin/zic
-yearistype=bin/yearistype
+zic=src/zic
+yearistype=src/yearistype
 output=zones-listing
 
 test -x $zic
@@ -14,7 +14,7 @@ rm -rf $output && mkdir -p $output
 
 zic_cmd="$zic -L /dev/null -y $yearistype "
 
-input="africa antarctica asia australasia europe northamerica southamerica iso8601"
+input="src/africa src/antarctica src/asia src/australasia src/europe src/northamerica src/southamerica src/iso8601"
 
 for continent in $input ; do
   test -f $continent
